src/structure/structure.ts

```diff
@@ -267,4 +357,258 @@ export class StructureViewer {
+        cell.style.gridRow = `${rowNum}`;
+
+        // this was removed because it conflicted with the buttons
+        // cell.onclick = () => {this._setActive(guid)}
+
+        // use HTML5 template to generate a DOM object from an HTML string
+        const activeFlag = document.createElement('div');
+        activeFlag.classList.add('chsp-inactive', 'chsp-circle-button');
+        activeFlag.id = `chsp-activate-${guid}`;
+        this._colors.push(getFarColor(this._colors));
+        activeFlag.style.backgroundColor = this._colors[this._colors.length - 1];
+        activeFlag.onclick = () => { this._setActive(guid); };
+        activeFlag.innerHTML = `<div><span class="tooltiptext">Choose as active</span></div>`;
+        cell.appendChild(activeFlag);
+
+        // use HTML5 template to generate a DOM object from an HTML string
+        const close = document.createElement('div');
+        close.classList.add('chsp-close', 'btn', 'btn-light', 'btn-sm');
+        close.id = `chsp-close-${guid}`;
+        close.onclick = () => {this._removeWidget(guid); this._resizeGridInc(-1); };
+        close.innerHTML = `<object>${close_svg}<span class="tooltiptext">Close widget</span></object>`;
+        cell.appendChild(close);
+
+        // use HTML5 template to generate a DOM object from an HTML string
+        const duplicate = document.createElement('div');
+        duplicate.classList.add('chsp-duplicate', 'btn', 'btn-light', 'btn-sm');
+        duplicate.id = `chsp-duplicate-${guid}`;
+        duplicate.onclick = () => {
+                this._resizeGridInc(1);
+                const myCurrent = this._currentList[this._guids.indexOf(guid)];
+                let index;
+                if (this._indexer.mode === 'structure') {
+                  index = this._indexer.from_structure_atom(myCurrent.structure);
+                  } else {
+                  index = this._indexer.from_structure_atom(myCurrent.structure,
+                                                            myCurrent.atom);
+                  }
+                this._currentList[this._currentList.length - 1] = myCurrent;
+                this.show(index, this._guids[this._guids.length - 1]);
+                };
+        duplicate.innerHTML = `<object>${duplicate_svg}<span class="tooltiptext">Add duplicate widget</span></object>`;
+        cell.appendChild(duplicate);
+
+        this._root.appendChild(cell);
+        } else {
+        cell.style.gridColumn = `${colNum}`;
+        cell.style.gridRow = `${rowNum}`;
+      }
+
+    }
+    /*
+     * Function to initialize the grid instance for nwidgets cells and place
+     * onto the DOM element mapped to `id`
+     */
+    private _setupGrid() {
+
+      // Determine best arrangement for nwidgets
+      const arrangement = bestArrangement(this._nwidgets);
+
+      if (this._widgets.length > this._nwidgets) {
+        sendWarning(`Warning: Eliminating last ${this._widgets.length - this._nwidgets} widgets.`);
```

Comment:
This is a bit strange from the user perspective. When would this occurs?

Reply:
I wanted to leave it open if we want to ad-hoc resize the grid with universal buttons. If there are more widgets than the new grid size, we'll eliminate the last few.